pool/hub_test.go

```diff
@@ -43,28 +219,175 @@ func testHub(t *testing.T, db *bolt.DB) {
 		MaxGenTime:            time.Second * 20,
 		PoolFeeAddrs:          []dcrutil.Address{poolFeeAddrs},
 		MaxTxFeeReserve:       maxTxFeeReserve,
-		MaxConnectionsPerHost: 2,
+		MaxConnectionsPerHost: 10,
 		NonceIterations:       iterations,
 		MinerPorts: map[string]uint32{
 			CPU:           5050,
 			InnosiliconD9: 5052,
 			AntminerDR3:   5553,
 			AntminerDR5:   5554,
 			WhatsminerD1:  5555,
+			ObeliskDCR1:   5551,
 		},
 	}
 	ctx, cancel := context.WithCancel(context.Background())
 	hub, err := NewHub(cancel, hcfg)
 	if err != nil {
-		t.Fatalf("[NewHub] uexpected error: %v", err)
+		t.Fatalf("[NewHub] unexpected error: %v", err)
 	}
 
+	notifHandlers := hub.CreateNotificationHandlers()
+	if notifHandlers == nil {
+		t.Fatalf("[CreatNotificationHandlers] expected an "+
+			"initialized notifications handler: %v", err)
+	}
+
+	// Create dummy wallet and node connections.
+	nodeConn := &tNodeConnection{}
+	hub.SetNodeConnection(nodeConn)
+	walletConn := &tWalletConnection{}
+	walletClose := func() error {
+		return nil
+	}
+	hub.SetWalletConnection(walletConn, walletClose)
 	err = hub.Listen()
 	if err != nil {
 		t.Fatalf("[Listen] uexpected error: %v", err)
 	}
+	err = hub.FetchWork()
+	if err != nil {
+		t.Fatalf("[FetchWork] unexpected error: %v", err)
+	}
 	go hub.Run(ctx)
 
+	// Create the mined work to be confirmed.
+	work := NewAcceptedWork(
+		"00008121c7731f9f81cae3d6279e81b9e7e7ebab94fb7bf584d16ecb70fbb9dd",
+		"000033925cfb136f209b2722c4149dd53fceb0323f74b39be753887c19edcd2c",
+		56,
+		"193c4b8fd02aaed33ab9c5418ace9bec4047f61f923767bceb5a51c6e368bfa6",
+		CPU)
+
+	err = work.Create(db)
+	if err != nil {
+		t.Fatalf("[Create] unexpected error: %v", err)
+	}
+
+	// Create pending payments for account X.
+	amt, err := dcrutil.NewAmount(5)
+	if err != nil {
+		t.Fatalf("[Create] unexpected error: %v", err)
+	}
+
+	pbx := makePaymentBundle(xID, 2, amt)
+	for _, pmt := range pbx.Payments {
+		err := pmt.Create(db)
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	// Create a client connection for each endpoint.
+	for _, e := range hub.endpoints {
+		_, err = net.Dial("tcp", e.listener.Addr().String())
+		if err != nil {
+			t.Fatalf("unexpected error: %v", err)
+		}
+	}
+
+	// Ensure the hub has connected clients.
+	time.Sleep(time.Millisecond * 200)
+	if !hub.HasClients() {
+		t.Fatal("expected hub to have connected clients")
+	}
+
+	// Ensure there are no connected clients for test accounts.
+	aInfo := hub.FetchAccountClientInfo(xID)
+	if len(aInfo) != 0 {
+		t.Fatalf("[FetchClientInfo] expected a client info size of 0"+
+			" for account x, got %d", len(aInfo))
+	}
+	aInfo = hub.FetchAccountClientInfo(yID)
+	if len(aInfo) != 0 {
+		t.Fatalf("[FetchClientInfo] expected a client info size of 0"+
+			" for account y, got %d", len(aInfo))
+	}
+
+	// Ensure there is a connected client for each endpoint.
+	cInfo := hub.FetchClientInfo()
+	if len(cInfo) == 0 {
```

Comment:
This doesnt actually check for one client on each endpoint - there could only be one client for one endpoint (and the others have none) and this test would still pass

Reply:
noted, rectifying.

Reply:
Please excuse the updated commit, I made the changes in place after rebasing. My bad.